wandb/__init__.pyi

```diff
@@ -416,48 +416,96 @@ def log(
-
     Logged values don't have to be scalars. Logging any wandb object is supported.
-    For example `wandb.log({"example": wandb.Image("myimage.jpg")})` will log an
+    For example `run.log({"example": wandb.Image("myimage.jpg")})` will log an
     example image which will be displayed nicely in the W&B UI.
     See the [reference documentation](https://docs.wandb.com/ref/python/data-types)
     for all of the different supported types or check out our
     [guides to logging](https://docs.wandb.ai/guides/track/log) for examples,
     from 3D molecular structures and segmentation masks to PR curves and histograms.
-    `wandb.Table`s can be used to logged structured data. See our
+    You can use `wandb.Table` to log structured data. See our
     [guide to logging tables](https://docs.wandb.ai/guides/data-vis/log-tables)
     for details.
 
     Logging nested metrics is encouraged and is supported in the W&B UI.
-    If you log with a nested dictionary like `wandb.log({"train":
-    {"acc": 0.9}, "val": {"acc": 0.8}})`, the metrics will be organized into
+    When you log with a nested dictionary like `run.log({"train":
+    {"acc": 0.9}, "val": {"acc": 0.8}})`, the metrics are organized into
     `train` and `val` sections in the W&B UI.
 
-    wandb keeps track of a global step, which by default increments with each
-    call to `wandb.log`, so logging related metrics together is encouraged.
-    If it's inconvenient to log related metrics together
-    calling `wandb.log({"train-loss": 0.5}, commit=False)` and then
-    `wandb.log({"accuracy": 0.9})` is equivalent to calling
-    `wandb.log({"train-loss": 0.5, "accuracy": 0.9})`.
+    `run.log` is not intended to be called more than a few times per second.
+    Instead, try logging once every N iterations, or collect data over multiple
+    iterations and log it in a single step.
+
+    ### Summary values
+
+    The UI shows summary values in the run overview page
+    and in the run table for comparison.
+    Summary values can be set directly by using the `summary` property:
+
+    ```
+    run.log({"train": {"loss": 100}})
+    run.log({"train": {"loss": 90}})
+    run.log({"train": {"loss": 98}})
+    run.summary["train"]["loss"] = 90
+    ```
+
+    See also [define_metric](https://docs.wandb.ai/ref/python/run#define_metric)
+    for more control over summary calculations.
+
+    ### The W&B step
+
+    With basic usage, each call to `log` creates a new "step".
+    The step must always increase, and it is not possible to log
+    to a previous step.
 
-    `wandb.log` is not intended to be called more than a few times per second.
-    If you want to log more frequently than that it's better to aggregate
-    the data on the client side or you may get degraded performance.
+    Note that you can use any metric as the X axis in charts.
+    In many cases, it is better to think of the W&B step as
+    a sort of timestamp rather than a training step.
```

Comment:
```suggestion
    a sort of logical counter rather than a training step.
```

what do you think about something like this?

Reply:
"logical counter" sounds pretty similar to "step" to me :) the point I'm trying to make is that a rule of thumb is that you shouldn't use a step where you wouldn't use a timestamp. I updated this text to say

> In many cases, it is better to treat the W&B step like you'd treat a timestamp rather than a training step.

what do you think?

Reply:
Oh so maybe logical clock. Something like lamport timestamp https://en.wikipedia.org/wiki/Lamport_timestamp

Reply:
I wouldn't want to open the can of works of how `step` works with distributed runs by mentioning Lamport timestamps :p what do you think of the updated wording though? "Like you'd treat a logical clock" doesn't sound as clear as "like you'd treat a timestamp" to me---runs have timestamps, but they don't have logical clocks.

Reply:
Perfect… yeah no need to mention lamport timestamp…